kong/pdk/request.lua

```diff
@@ -98,7 +96,7 @@ local function new(self)
   function _REQUEST.get_scheme()
     check_phase(PHASES.request)
 
-    return var.scheme
+    return ngx.ctx.scheme or var.scheme
```

Comment:
`ctx` is cheaper then `var`, so use `ctx` if exists, like what we do in router handling: `local req_scheme = ctx and ctx.scheme or var.scheme`